membershiprelation.php

```diff
@@ -52,6 +52,37 @@ function membershiprelation_civicrm_post($op, $objectName, $objectId, &$objectRe
     }
   }
 
+  function membershiprelation_civicrm_pre($op, $objectName, $objectId, &$params) {
+    if ($op == "edit" && $objectName == "Membership") {
+      if (!empty($params['membership_type_id'])) {
+        if (in_array($params['membership_type_id'], [VMTONEGIRL, VMTTWOGIRLS, VMTTHREEGIRLS, VMTFOURGIRLS])) {
+          $membershipObj = new CRM_Member_DAO_Membership();
+          $membershipObj->id = $objectId;
+          $membershipObj->find();
+          $oldType = NULL;
+          while ($membershipObj->fetch()) {
+            $oldType = $membershipObj->membership_type_id;
+          }
+          if (!in_array($oldType, [VMTONEGIRL, VMTTWOGIRLS, VMTTHREEGIRLS, VMTFOURGIRLS])) {
+            unset($params['id']);
+          }
+        }
+        else {
+          $membershipObj = new CRM_Member_DAO_Membership();
+          $membershipObj->id = $objectId;
+          $membershipObj->find();
+          $oldType = NULL;
+          while ($membershipObj->fetch()) {
+            $oldType = $membershipObj->membership_type_id;
+          }
+          if (in_array($oldType, [VMTONEGIRL, VMTTWOGIRLS, VMTTHREEGIRLS, VMTFOURGIRLS])) {
+            unset($params['id']);
```

Comment:
hmm What if they are just doing an edit on an existing member e.g. changing status? isn't this problematic in that sense?

Reply:
In that case, the $oldType is not a VM for sure because its in else part. For status change the old and new memebership type must be same and thus it won't fall under the 2nd loop inside the if/else part

Reply:
@monishdeb What if they are doing an edit to a VM membership e.g. edit source or something else?, It seems to me that we could land here if the membership type is a VM and is already a VM or maybe i'm not fully reading the code correctly

Reply:
so if someone doing edit on VM then it satisfy IF considtion at L58 above. Then next it fetches the membership object on the basis of $id and it founds that the $oldType is same i.e. VM. Then it doesn't fulfill the second loop i.e. at L66 and thus it won't unset the membershipID, and thus allowing the membership object to update.

Reply:
Well if somepne changes the VM type either from backoffice or online form it creates a new membership of the new type and vice-versa holds true too.